cloud/azure/deploy/api/apimanagement.go

```diff
@@ -239,6 +245,41 @@ func NewAzureApiManagement(ctx *pulumi.Context, name string, args *AzureApiManag
 		}
 	}
 
+	if args.Cors != nil {
+		corsConfig, err := cors.GetCorsConfig(args.Cors)
+		if err != nil {
+			return nil, err
+		}
+
+		corsTemplate := fmt.Sprintf(`<cors allow-credentials="%s">`, strconv.FormatBool(corsConfig.AllowCredentials))
+
+		// origins
+		corsTemplate = corsTemplate + addCorsElements(corsConfig.AllowOrigins, "allowed-origins", "origin", "")
+
+		// methods
+		corsTemplate = corsTemplate + addCorsElements(corsConfig.AllowMethods, "allowed-methods", "method", fmt.Sprintf(`preflight-result-max-age="%s"`, strconv.FormatInt(int64(corsConfig.MaxAge), 10)))
+
+		// headers
+		corsTemplate = corsTemplate + addCorsElements(corsConfig.AllowHeaders, "allowed-headers", "header", "")
+
+		// expose headers
+		corsTemplate = corsTemplate + addCorsElements(corsConfig.ExposeHeaders, "expose-headers", "header", "")
+
+		corsTemplate = corsTemplate + "</cors>"
```

Comment:
Would embedding a document with position directives make this a bit easier to read and follow in future?

We could include this in a separate file in the package e.g `cors.go` with a method that has args for each of the elements it can template.

If we need more complexity we can also include a proper templating engine in future as well.

Reply:
Yeah it would, lets do that.

Reply:
switched to using templating, since it needs to be dynamic. Resolved